src/node_modules/app/machines/builder/form.options.js

```diff
@@ -159,6 +223,50 @@ function computeDefaultValues (ctx) {
 	}
 }
 
+function processRequestValue (value, type) {
+	if (value === undefined) {
+		return undefined;
+	}
+	if (is_native(type)) {
+		return value;
+	}
+	if (is_intWithUnit(type)) {
+		return `${value.value}${value.unit}`;
+	}
+	if (is_object(type) || is_extent(type) || is_xor(type)) {
+		const outgoing = {};
+		const typings = getTypings(type);
+		typings.forEach(typing => {
+			let key = typing.name;
+			let obj = value[key];
+			if (obj === undefined) {
+				return;
+			}
+			let currentType = typing.type;
+			if (is_xor(currentType)) {
+				key = obj.__selection;
+				obj = obj[key];
+				currentType = currentType[key];
+			}
+			const result = processRequestValue(obj, currentType);
+			if (
+				isNotNil(result)
+				&& is_native(currentType)
+				|| isNotNil(result)
+				&& isObjNotEmpty(result)
+				// type is optional with empty object default (existence)
+				|| is_optional(currentType)
+				&& hasDefault(currentType)
+				&& isObjEmpty(getDefault(currentType))
+			) {
+				outgoing[key] = result;
+			}
+		});
```

Comment:
- I'd rename `outgoing` to `params`
- I'd use `reduce`
- instead of `typing => {` I'd use deconstructing